logstash-core/lib/logstash/pipeline.rb

```diff
@@ -46,6 +47,7 @@ def initialize(config_str, settings = SETTINGS)
     @inputs = nil
     @filters = nil
     @outputs = nil
+    @execution_context = LogStash::ExecutionContext.new(pipeline_id)
```

Comment:
maybe `@pipeline_id` would be better? I know the `attr_reader` is kicking in here, but it would maybe be easier to read